src/egraph.rs

```diff
@@ -65,6 +65,20 @@ impl<L: Language> EClass<L> {
         self.id = id;
         self.nodes = more_nodes;
     }
+
+    pub fn get_constant(&self) -> Option<RecExpr<L>> {
```

Comment:
If you change the type of `eval` (see below), then this should return `Option<L::Constant>`.